var/spack/repos/builtin/packages/py-statsmodels/package.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyStatsmodels(PythonPackage):
+    """Statistical computations and models for use with SciPy"""
+
+    homepage = "http://www.statsmodels.org"
+    url      = "https://pypi.io/packages/source/s/statsmodels/statsmodels-1.9.1.tar.gz"
```

Comment:
The URL "works", as in Spack correctly extrapolates the version and downloads the package. But if you take a look at #2972, I'm trying to make sure that Spack can correctly extrapolate the version based on the URL. This relies on some means of knowing whether or not the version was correctly parsed. If I have a URL like "https://pypi.io/packages/source/s/statsmodels/statsmodels-1.9.1.tar.gz", how do I know if the correct answer is `1.9.1`? The way things work is that `spack url` checks whether or not version `1.9.1` is present in the package. If it is, we know we correctly parsed the version. If not, we have no guarantee. So yes, while the URL can be extrapolated correctly, using a URL for a version that is not in Spack causes the `spack url` command to be less useful.

> is it intentional that 0.8.0rc1 takes precedent?

Yes. See https://github.com/LLNL/spack/blob/develop/lib/spack/spack/version.py#L264 for details. Looks like that's how RPM sorts versions, so we went with it. Perhaps this isn't the right thing to do, as alpha, beta, and release candidates are generally older than the non-letter releases. But that's an argument for another day. If you want `0.8.0` to take precedence, you can either remove `0.8.0rc1` or add `preferred=True` to `0.8.0`. Is there any reason you added `0.8.0rc1`? We don't generally add release candidates, as they go through less testing than actual releases and are often buggy.